src/osd/OSD.cc

```diff
@@ -11140,7 +11140,7 @@ void OSD::ShardedOpWQ::stop_for_fast_shutdown()
 {
   m_fast_shutdown = true;
 
-  for (int shard_index = 0; shard_index < osd->num_shards; shard_index++) {
+  for (uint64_t shard_index = 0; shard_index < osd->num_shards; shard_index++) {
```

Comment:
In https://github.com/ceph/ceph/pull/53530, @ifed01 used `size_t` instead of `uint64_t`. The difference is that `size_t` can store the maximum size of an object of any type, as opposed to `uint64_t`, which can fail if the index exceeds `UINT_MAX`.

wdyt?

Reply:
I don't know of a machine where `size_t` exceeds `uint64_t`. `size_t` doesn't seem appropriate for a shard count in any case as its a measure of an object size (generally as returned by `malloc`).

Reply:
IMO in this case it doesn't matter much. Personally I prefer to avoid explicit integer size specification when there is no real need for it. But that's not a big deal anyway.

Reply:
In fact it looks like we deal with a pretty limited amount of shards at this point. I don't believe it ever exceeds event 16-bit width.  Hence again that's not a big deal which type to use here..